lib/commands/os/versions.ts

```diff
@@ -50,16 +54,20 @@ export default class OsVersionsCmd extends Command {
 
 	public static flags: flags.Input<FlagsDef> = {
 		help: cf.help,
+		esr: flags.boolean({
+			description: 'select balenaOS ESR versions',
+			default: false,
+		}),
```

Comment:
In my first go at this PR, I did not have a `--esr` flag and all versions were listed together. But some issues mentioned for `os download` also applied here:

* Two versions labeled "recommended", one ESR and one non-ESR. We could amend the formatting, but:
  * If we didn't print "recommended" at all, it would strictly be a breaking change (re output) for non-ESR versions.
  * If we printed "recommended" only for non-ESR versions, then... are we implying that ESR versions are not recommended? :-)
  * If we did some other special formatting thing for `os versions`, then the formatting code could not be shared with the interactive menu of `os download`.

* Re sorting, having ESR first and non ESR later or vice versa, there was still an element of the issue mentioned in a comment for the `os download` command: whatever we list at the top _with a recommended label_ is not necessarily the best (actually recommended) option for the user. With a `--esr` flag, the "recommendation" always fits the user's choice of ESR or non ESR.

* And finally, the web dashboard has a separate dropdown box to select ESR versions, avoiding listing them together. That solution has a 1-1 mapping to a `--esr` flag, which allows me to claim "platform UI/UX consistency" as well. :-)